hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -298,6 +247,19 @@ public StorageStatistics provide() {
         LOG.debug("Using metadata store {}, authoritative={}",
             getMetadataStore(), allowAuthoritative);
       }
+
+      ExecutorService downloadExecutorService = Executors.newFixedThreadPool(16, new ThreadFactoryBuilder().setNameFormat("multipart-download-%d").build());
```

Comment:
might be worth pulling these out into final instance variables so they aren't magic numbers

Reply:
Decided to make them configurable via hadoop conf instead.